pyxem/signals/diffraction2d.py

```diff
@@ -987,48 +978,19 @@ def center_of_mass(
 
         """
         det_shape = self.axes_manager.signal_shape
-        nav_dim = self.axes_manager.navigation_dimension
-        if chunk_calculations is None:
-            chunk_calculations = [16] * nav_dim + list(det_shape)
         if mask is not None:
             x, y, r = mask
-            mask_array = pst._make_circular_mask(x, y, det_shape[0], det_shape[1], r)
-            mask_array = np.invert(mask_array)
-        else:
-            mask_array = None
-        if self._lazy:
-            dask_array = self.data.rechunk(chunk_calculations)
-        else:
-            dask_array = da.from_array(self.data, chunks=chunk_calculations)
-        data = dt._center_of_mass_array(
-            dask_array, threshold_value=threshold, mask_array=mask_array
+            mask = pst._make_circular_mask(x, y, det_shape[0], det_shape[1], r)
+
+        ans = self.map(
+            center_of_mass, threshold=threshold, mask=mask, inplace=inplace, **kwargs
         )
-        if lazy_output:
-            if nav_dim == 2:
-                s_com = LazyDPCSignal2D(data)
-            elif nav_dim == 1:
-                s_com = LazyDPCSignal1D(data)
-            elif nav_dim == 0:
-                s_com = LazyDPCBaseSignal(data).T
-        else:
-            if show_progressbar:
-                pbar = ProgressBar()
-                pbar.register()
-            data = data.compute()
-            if show_progressbar:
-                pbar.unregister()
-            if nav_dim == 2:
-                s_com = DPCSignal2D(data)
-            elif nav_dim == 1:
-                s_com = DPCSignal1D(data)
-            elif nav_dim == 0:
-                s_com = DPCBaseSignal(data).T
-        s_com.axes_manager.navigation_axes[0].name = "Beam position"
-        for nav_axes, sig_axes in zip(
-            self.axes_manager.navigation_axes, s_com.axes_manager.signal_axes
-        ):
-            pst._copy_axes_object_metadata(nav_axes, sig_axes)
-        return s_com
+        if inplace:
+            ans = self
+        ans = ans.T
+        ans.set_signal_type("dpc")
+        ans.axes_manager.navigation_axes[0].name = "Beam position"
+        return ans
```

Comment:
This approach is poorly behaved for the inplace case:

![image](https://github.com/pyxem/pyxem/assets/32455330/b2b39fa9-de2a-4de1-ba49-b001644de8bb)

Specifically, (1) it returns and object and (2) the inplace doesn't get it's dimension fixed quite correctly.

Reply:
original signal was 10,10,20,20 in this dummy example run off the PR at present.

Reply:
@pc494 Wow I was very confidently incorrect about how that works.  The joys of python.... Apparently if you operate on the signal afterwards (with the transpose) it will deepcopy the array and you no longer are pointing to the original.

Reply:
Trust is a dangerous thing when you're writing Python code...